.github/workflows/release.yml

```diff
@@ -0,0 +1,100 @@
+name: Release
+
+on:
+  push:
+    tags:
+      - v**
+
+jobs:
+  build:
+    name: 'Build Assets'
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        build: [linux_amd64, windows_amd64, darwin_amd64, darwin_arm64]
+        compress: [tar]
+        include:
+          - build: linux_amd64
+            goos: linux
+            goarch: amd64
+            ext: ''
+          - build: windows_amd64
+            goos: windows
+            goarch: amd64
+            ext: '.exe'
+            compress: zip
+          - build: darwin_amd64
+            goos: darwin
+            goarch: amd64
+            ext: ''
+          - build: darwin_arm64
+            goos: darwin
+            goarch: arm64
+            ext: ''
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: "1.20"
+          check-latest: true
+      - name: Build
+        run: |
+          GOOS=${{ matrix.goos }} GOARCH=${{ matrix.goarch }} go build cmd/autometrics/main.go
+          mv main${{ matrix.ext }} autometrics${{ matrix.ext }}
+
+      - name: Pack (Tar)
+        if: matrix.compress == 'tar'
+        run: |
+          tar -c -f autometrics-${{ matrix.build }}.tar.gz -v -z autometrics${{ matrix.ext }}
+      - name: Upload (Tar)
+        if: matrix.compress == 'tar'
+        uses: actions/upload-artifact@v3
+        with:
+          name: autometrics ${{ matrix.build }}
+          path: autometrics-${{ matrix.build }}.tar.gz
+          if-no-files-found: error
+          retention-days: 1
```

Comment:
Actually, both mac and linux support `.zip` compression so we can maybe get rid of the in-action compression altogether and just let GitHub artifacts do its own `.zip` compression

Reply:
Zip loses more file permissions than tarballing, I didn't want to risk losing the executable permission. I'll try to find out whether zip is usable for `755` files on unix though

Reply:
if we keep compression anyways (i agree we should thanks to your reasoning above) we can keep them as `.tar.gz`. `.zip` does support unix file perms with a external attribute, which is supported by e.g `unzip` and 7Zip but idk how the most widespread unzip programs on linux/mac so i couldn't say if they do. safest to go with `.tar.gz`

Reply:
I just tested on the mac (a zip->unzip loop) and it kept the executable perm, so I'll just use zip everywhere